server/apps/immich/src/middlewares/admin-role-guard.middleware.ts

```diff
@@ -1,42 +1,12 @@
-import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
-import { Reflector } from '@nestjs/core';
-import { InjectRepository } from '@nestjs/typeorm';
-import { Repository } from 'typeorm';
-import { UserEntity } from '@app/database/entities/user.entity';
-import { ImmichJwtService } from '../modules/immich-jwt/immich-jwt.service';
+import { CanActivate, ExecutionContext, Injectable, UseGuards } from '@nestjs/common';
+import { Request } from 'express';
 
 @Injectable()
 export class AdminRolesGuard implements CanActivate {
-  constructor(
-    private reflector: Reflector,
-    private jwtService: ImmichJwtService,
-    @InjectRepository(UserEntity)
-    private userRepository: Repository<UserEntity>,
-  ) {}
-
   async canActivate(context: ExecutionContext): Promise<boolean> {
-    const request = context.switchToHttp().getRequest();
-    let accessToken = '';
-
-    if (request.headers['authorization']) {
-      accessToken = request.headers['authorization'].split(' ')[1];
-    } else if (request.cookies['immich_access_token']) {
-      accessToken = request.cookies['immich_access_token'];
-    } else {
-      return false;
-    }
-
-    const { userId } = await this.jwtService.validateToken(accessToken);
-
-    if (!userId) {
-      return false;
-    }
-
-    const user = await this.userRepository.findOne({ where: { id: userId } });
-    if (!user) {
-      return false;
-    }
-
-    return user.isAdmin;
+    const request = context.switchToHttp().getRequest<Request>();
+    return request.user?.isAdmin || false;
```

Comment:
This always returns `undefined`

Reply:
The easiest way to test this condition is when you access the admin page on the web